src/testutils.rs

```diff
@@ -437,6 +439,60 @@ impl DatabaseConnection for DummyDatabase {
         txids_and_time.truncate(limit as usize);
         txids_and_time.into_iter().map(|(txid, _)| txid).collect()
     }
+
+    fn list_missing_txids(&mut self) -> Vec<bitcoin::Txid> {
+        let txs = self.db.read().unwrap().txs.clone();
+        let coins = self.coins(&[], &[]);
+        // Get deposit and spend txids of all coins and keep only those not in txs.
+        coins
+            .keys()
+            .map(|op| op.txid)
+            .chain(coins.values().filter_map(|c: &Coin| c.spend_txid))
+            .filter(|txid| !txs.contains_key(txid))
+            .collect()
+    }
```

Comment:
Note this may currently return duplicate txids. Perhaps I should rather leave this as `todo!()` as the code is not currently used anywhere.

Reply:
This code is indeed used by the tests so instead I changed it to remove duplicate txids.